src/common/list-syntax.k

```diff
@@ -0,0 +1,311 @@
+syntax TypeParamList ::= List{TypeParam,","} [klabel('TypeParamList)]
+syntax AbstractMethodModList ::= List{AbstractMethodMod, ""} [klabel('AbstractMethodModList)]
+syntax AnnoAbstractMethodModList ::= List{AnnoAbstractMethodMod, ""} [klabel('AnnoAbstractMethodModList)]
+syntax AnnoMethodModList ::= List{AnnoMethodMod, ""} [klabel('AnnoMethodModList)]
+syntax AnnoVarModList ::= List{AnnoVarMod, ""} [klabel('AnnoVarModList)]
+syntax AnnoClassModList ::= List{AnnoClassMod, ""} [klabel('AnnoClassModList)]
+syntax AnnoConstrModList ::= List{AnnoConstrMod, ""} [klabel('AnnoConstrModList)]
+syntax AnnoConstantModList ::= List{AnnoConstantMod, ""} [klabel('AnnoConstantModList)]
+syntax AnnoFieldModList ::= List{AnnoFieldMod, ""} [klabel('AnnoFieldModList)]
+syntax AnnoInterfaceModList ::= List{AnnoInterfaceMod, ""} [klabel('AnnoInterfaceModList)]
+syntax AnnoList ::= List{Anno, ""} [klabel('AnnoList)]
+syntax AnnoElemDecList ::= List{AnnoElemDec, ""} [klabel('AnnoElemDecList)]
+syntax InterfaceMemberDecList ::= List{InterfaceMemberDec, ""} [klabel('InterfaceMemberDecList)]
+syntax ElemValPairList ::= List{ElemValPair,","} [klabel('ElemValPairList)]
+syntax ElemValList ::= List{ElemVal,","} [klabel('ElemValList)]
+syntax StringPartList ::= List{StringPart,""} [klabel('StringPartList)]
+syntax EnumConstList ::= List{ EnumConst, ","} [klabel(' EnumConstList)]
+syntax ClassBodyDecList ::= List{ClassBodyDec, ""} [klabel('ClassBodyDecList)]
+syntax BlockStmList ::= List{BlockStmt, ""} [klabel('BlockStmList)]
+syntax SwitchGroupList ::= List{SwitchGroup, ""} [klabel('SwitchGroupList)]
+syntax SwitchLabelList ::= List{SwitchLabel, ""} [klabel('SwitchLabelList)]
+
+
+ syntax UserList ::=   BlockStmList
+                      | SwitchGroupList
+                      | SwitchLabelList
+                      | InterfaceTypeList
+                      | ExceptionTypeList
+                      | IdList
+                      | TypeDecList
+                      | VarDecList
+                      | ImportDecList
+                      | ActualTypeArgList
+                      | ClassOrInterfaceTypeList
+                      | TypeParamList
+                      | AbstractMethodModList
+                      | AnnoAbstractMethodModList
+                      | AnnoMethodModList
+                      | AnnoVarModList
+                      | AnnoClassModList
+                      | AnnoConstrModList
+                      | AnnoConstantModList
+                      | AnnoFieldModList
+                      | AnnoInterfaceModList
+                      | AnnoList
+                      | AnnoElemDecList
+                      | InterfaceMemberDecList
+                      | ElemValPairList
+                      | ElemValList
+                      | StringPartList
+                      | EnumConstList
+                      | ClassBodyDecList
+
+
+                      syntax   InterfaceType
```

Comment:
I checked again, I think the UserList syntax definition is fine, so you mean why 
                      syntax   InterfaceType
                      syntax   ExceptionType
                      syntax   Id
                      syntax   TypeDec::= ClassDec
                      syntax   VarDec
                      syntax   ImportDec
                      syntax   ActualTypeArg
                      syntax   ClassOrInterfaceType
                      syntax   TypeParam
                      syntax   AbstractMethodMod
                      syntax   AnnoAbstractMethodMod
                      syntax   AnnoMethodMod
                      syntax   AnnoVarMod
                      syntax   AnnoClassMod
                      syntax   AnnoConstrMod
                      syntax   AnnoConstantMod
                      syntax   AnnoFieldMod
                      syntax   AnnoInterfaceMod
                      syntax   Anno
                      syntax   AnnoElemDec
                      syntax   InterfaceMemberDec
                      syntax   ElemValPair
                      syntax   ElemVal
                      syntax   StringPart
                      syntax   EnumConst
                      syntax   ClassBodyDec

are indented more, right?
Then as I explained, just put there to remind me to remove.